### PR TITLE
Add initial crate

### PR DESCRIPTION
**Pull request by leighmcculloch**
### What

Add initial crate and npm package building, from:
- https://github.com/stellar/rs-stellar-xdr/pull/351
- https://github.com/stellar/rs-stellar-xdr/pull/360

### Why

For exporting the XDR<>JSON functionality from the stellar-xdr Rust crate for use in the laboratory, and elsewhere.

### CODE REVIEW

File: src/utils.rs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+// #[allow(dead_code)]
```
Comment (willemneal) [inline src/utils.rs]: This was the default. Why the removal and why a comment instead of delete?

Comment (leighmcculloch) [reply src/utils.rs]: Oops I meant to uncomment this. I'll do that now. Thanks.

Comment (leighmcculloch) [reply src/utils.rs]: 804ee20

**Comment by leighmcculloch** [comment]
TODO:
- [ ] Add basic tests.
    - [ ] Figure out how to use wasm-bindgen-test without needing to add rlib to the Cargo.toml that causes disables LTO and makes the wasm bigger
    - [ ] Get geckodriver working
    - [ ] Add a test for the types function
- [ ] Update contributing docs to include what tools to install (wasm-pack, wasm-bindgen-cli, geckodriver), etc

**Review by willemneal — APPROVED** [review]
Looks great! Glad we can final get something a least a little stable out of this.

**Pull request merged**